dill/_dill.py

```diff
@@ -504,29 +505,138 @@ def dump_session(filename='/tmp/session.pkl', main=None, byref=False, **kwds):
         pickler._main_modified = main is not pickler._original_main
         pickler.dump(main)
     finally:
-        if f is not filename:  # If newly opened file
-            f.close()
+        if file is not filename:  # if newly opened file
+            file.close()
     return
 
-def load_session(filename='/tmp/session.pkl', main=None, **kwds):
-    """update the __main__ module with the state from the session file"""
-    if main is None: main = _main_module
+def _open_peekable(filename):
```

Comment:
The point of this function is to try to ensure the file is in a mode where one can `peek` without advancing?